README.md

```diff
@@ -112,11 +112,11 @@ Effects are run with _effect handlers_, specified as functions (generally starti
 ```haskell
 example1 :: (Carrier sig m, Effect sig) => [a] -> m (Int, ())
 example1 list = runState 0 $ do
-  i <- get
+  i <- get @Int
   put (i + length list)
 ```
 
-`runState` returns a tuple of both the computed value (the `()`) and the final state (the `Int`), visible in the result of the returned computation.
+`runState` returns a tuple of both the computed value (the `()`) and the final state (the `Int`), visible in the result of the returned computation. The `get` function is resolved with a visible type application, due to the fact that effects can contain more than one state type (in contrast with `mtl`'s `MonadState`, which limits the user to a single state type).
```

Comment:
I like this, but it sort of gives the impression that it’s mandatory, which isn’t the case when it can be resolved by inference (like in this example, due to the type of `length`). Not sure whether that warrants another example or not, tbh.